### PR TITLE
handles multiple persons list items

### PR DESCRIPTION
**Pull request by HiteshRepo**
handles multiple persons list items.

**Original `person` list with single value**:
Person-List
 
**Restored `person` list with single value**:
Restored-Person-List

**Original `person` list with multi value**:
Person-List-Multi

**Restored `person` list with multi value**:
Restored-Person-List-Multi

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)
#5108 
#5084 

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/pkg/services/m365/api/lists.go [hunk 1]
```diff
@@ -380,8 +386,18 @@ func setColumnType(
...
 ) {
+	colName := ptr.Val(newColumn.GetName())
 	colDetails := &columnDetails{}
 
+	// for certain columns like 'person', the column name is for example 'personName'
+	// if the list item for that column holds single value,
+	// the field data is fetched as '{"personNameLookupId": "10"}'
+	// if the list item for that column holds multiple values,
+	// the field data is fetched as '{"personName": [{"lookupId": 10}, {"lookupId": 11}]}'.
+	// Hence this is function helps us to determine which name to use while accessing stored data
```
Comment (pandeyabs) [inline src/pkg/services/m365/api/lists.go:397]:
```suggestion
	// Hence this function helps us to determine which name to use while accessing stored data
```

File: src/pkg/services/m365/api/lists.go [hunk 2]
```diff
@@ -490,20 +518,57 @@ func setAdditionalDataByColumnNames(
...
 
-		specifyODataType(filteredData, colDetails, colName)
+		specifyODataType(filteredData, colDetails, colDetails.createFieldName)
 	}
 
 	return filteredData
 }
 
+func populateMultipleValues(val any, filteredData map[string]any, colDetails *columnDetails) {
+	if !colDetails.isMultipleEnabled {
```
Comment (pandeyabs) [inline src/pkg/services/m365/api/lists.go:535]: nit: `isMultipleEnabled` and `isPersonColumn` checks can be folded into a single if block

**Review by pandeyabs — APPROVED** [review]
🚢

File: src/pkg/services/m365/api/lists.go (hunk 2 above)
Comment (HiteshRepo) [reply src/pkg/services/m365/api/lists.go:535]: yeah. In next PR I was going to add for lookupColumn check. Hence wanted to keep checks separate for their corresponding reasoning.

File: src/pkg/services/m365/api/lists.go [hunk 3]
```diff
@@ -490,20 +518,57 @@ func setAdditionalDataByColumnNames(
...
+		return
+	}
+
+	if !colDetails.isPersonColumn {
+		return
+	}
+
+	multiNestedFields, ok := val.([]any)
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/api/lists.go:543]: You could extend he the `isSlice` check recommended from the other PR here to have a return value like `([]T, bool)`.  That'd be a more centralized way of parsing slices than you have here.  If you do that, I'd sugges moving the func to `common/tform`.

File: src/pkg/services/m365/api/lists.go [hunk 4]
```diff
@@ -490,20 +518,57 @@ func setAdditionalDataByColumnNames(
...
+	if !ok || len(multiNestedFields) == 0 {
+		return
+	}
+
+	lookupIDs := make([]float64, 0)
+
+	for _, nestedFields := range multiNestedFields {
+		if md, ok := nestedFields.(map[string]any); ok && keys.HasKeys(md,
+			[]string{LookupIDKey, LookupValueKey, PersonEmailKey}...) {
+			lookupID, ok := md[LookupIDKey].(*float64)
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/api/lists.go:553]: Nit: this conditional is trying to do too much on one line and that makes it difficult to read.
```suggestion
	lookupIDs := make([]float64, 0)
	lookupKeys := []string{LookupIDKey, LookupValueKey, PersonEmailKey}

	for _, nestedFields := range multiNestedFields {
		md, ok := nestedFields.(map[string]any)
		if !ok || !keys.HasKeys(lookupKeys...){
			continue
		}
		
		lookupID, ok := md[LookupIDKey].(*float64)
```

Comment (ryanfkeepers) [inline src/pkg/services/m365/api/lists.go:553]: this transform to a float64 should use the common/tform package (github won't let me suggest only line 553, so just putting the example here).
```suggestion
	v, err := tform.AnyValueToT[float64](lookupIDKey, md)
```

**Review by ryanfkeepers — COMMENTED** [review]
some implementation nits.

**Pull request merged**